lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -0,0 +1,398 @@
+              `null` in case the response contains the first page of results.
+
+        **Example GET Response**
+
+            {
+                "count": 99,
+                "next": "https://openedx-ccx-api-instance.org/api/ccx/v0/ccx/?course_id=<course_id>&page=2",
+                "previous": null,
+                "results": {
+                    {
+                        "ccx_course_id": "ccx-v1:Organization+EX101+RUN-FALL2099+ccx@1",
+                        "display_name": "CCX example title",
+                        "coach_email": "john@example.com",
+                        "start": "2019-01-01",
+                        "due": "2019-06-01",
+                        "max_students_allowed": 123
+                    },
+                    { ... }
+                }
+            }
+
+        **POST Response Values**
+
+            If the request for the creation of a CCX Course is successful, an HTTP 201 "Created" response
+            is returned with the newly created CCX details.
+
+            The HTTP 201 response has the following values.
+
+            * ccx_course_id: A string representation of a CCX Course ID.
+
+            * display_name: A string representing the CCX Course title.
+
+            * coach_email: A string representing the CCX owner email.
+
+            * start: A string representing the start date for the CCX Course.
+
+            * due: A string representing the due date for the CCX Course.
+
+            * max_students_allowed: An integer representing he maximum number of students that
+              can be enrolled in the CCX Course.
+
+        **Example POST Response**
+
+            {
+                "ccx_course_id": "ccx-v1:Organization+EX101+RUN-FALL2099+ccx@1",
+                "display_name": "CCX example title",
+                "coach_email": "john@example.com",
+                "start": "2019-01-01",
+                "due": "2019-06-01",
+                "max_students_allowed": 123
+            }
+    """
+    serializer_class = CCXCourseSerializer
+    pagination_class = CCXAPIPagination
+
+    def get(self, request):
+        """
+        Gets a list of CCX Courses for a given Master Course.
+
+        Additional parameters are allowed for pagination purposes.
+
+        Args:
+            request (Request): Django request object.
+
+        Return:
+            A JSON serialized representation of a list of CCX courses.
+        """
+        master_course_id = request.GET.get('master_course_id')
+        if master_course_id is None:
+            log.info('Master course ID not provided')
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'error_code': 'master_course_id_not_provided'
+                }
+            )
+        # the master course ID string has been escaped because passed as URI parameter
+        master_course_id = master_course_id.replace(' ', '+')  # TODO: find a better way for this
```

Comment:
I know, so should I assume it is encoded? because if it is in the url like
```
http://192.168.33.10:8000/api/ccx/v0/ccx/ccx-v1:Organization+CN101+RUN-FALL2015-AGAIN+ccx@5
```
the variable is totally fine (see line 315 of this file) and does not need to be encoded 

then it becomes a bit tricky and unclear how to build URLs on the client